src/dotnet-interactive.Tests/StdIoKernelConnectorTests.cs

```diff
@@ -135,6 +162,81 @@ public async Task it_throws_if_there_is_no_remote_subkernel_with_the_specified_n
             await action.Should().ThrowAsync<InvalidOperationException>();
         }
 
+        [Fact]
+        public async Task value_sharing_between_remote_kernels_works()
+        {
+            var connector = CreateConnector();
+            using var rootProxyKernel = await connector.CreateRootProxyKernelAsync();
+
+            var result = await rootProxyKernel.SendAsync(new RequestKernelInfo());
+            var kernelInfos = result.Events.OfType<KernelInfoProduced>().Select(e => e.KernelInfo);
+
+            using var _ = new AssertionScope();
+
+            var csharpKernelInfo = kernelInfos.Should().ContainSingle(i => i.LanguageName == "C#").Which;
+            using var csharpProxyKernel = await connector.CreateProxyKernelAsync(remoteInfo: csharpKernelInfo);
+            csharpProxyKernel.UseValueSharing();
+
+            var fsharpKernelInfo = kernelInfos.Should().ContainSingle(i => i.LanguageName == "F#").Which;
+            using var fsharpProxyKernel = await connector.CreateProxyKernelAsync(remoteInfo: fsharpKernelInfo);
+            fsharpProxyKernel.UseValueSharing();
+
+            using var composite = new CompositeKernel { csharpProxyKernel, fsharpProxyKernel };
+            composite.DefaultKernelName = csharpProxyKernel.Name;
+
+            result = await composite.SendAsync(new SubmitCode("let x = 123", "fsharp"));
+            result.Events.Last().Should().BeOfType<CommandSucceeded>();
+
+            result = await composite.SendAsync(new SubmitCode("#!set --name y --value @fsharp:x", "csharp"));
+            result.Events.Last().Should().BeOfType<CommandSucceeded>();
+
+            result = await composite.SendAsync(new SubmitCode("#!share --from fsharp x", "csharp"));
+            result.Events.Last().Should().BeOfType<CommandSucceeded>();
+
+            result = await composite.SendAsync(new SubmitCode("x + y", "csharp"));
+            result.Events.Last().Should().BeOfType<CommandSucceeded>();
+            var returnValueProduced = result.Events.OfType<ReturnValueProduced>().Should().ContainSingle().Which;
+            returnValueProduced.FormattedValues.Should().ContainSingle().Which.Value.Should().Contain("246");
```

Comment:
This can be chained. Ending the line with `.Which` is kind of confusing to read.